script.js

```diff
@@ -0,0 +1,70 @@
+// const currencyRate = {
+//   timestamp: 1620386223,
+//   base: "DKK",
+//     rates: {
+//         USD: 0.1444,
+//         EUR: 0.134,
+//         GBP: 0.1151,
+//         CAD: 0.1963,
+//         JPY: 20.9142,
+//         DKK: 1,
+//     },
+// };
+
+let currencyRate = {
+  timestamp: 1519296206, 
+  base: "EUR",
+  date: "2023-12-12", 
+  rates: {},
+};
+
+
+const rates = currencyRate.rates;
+console.log(rates);
+
+// const currOptions = document.getElementById("curr_output");
+// rates.forEach(element => {
+//      console.log(element);
+// });
+
+function addRate() {
```

Comment:
This is really great idea! I like it but unfortunately, it creates duplicates of the previous values every time

<img width="367" alt="Screenshot 2023-12-17 at 12 00 33" src="https://github.com/AlesyaMazurenko/currency-converter/assets/48878547/14409495-d0a8-4353-b046-5f7c9cc02991">

Reply:
There are multiple ways how to address this issue but let me show how in my opinion it could look like 

1. first in the `onRateSubmit` function you would check whether or not the `currencyRate` is already in the rates `object`
 ```
 if (currBase !== '' && newCurrency !== '') {
 // here we are checking if the currency has been already added, if not lets add it as a option via addRate
   if (!currencyRate.rates[newCurrency]) {
      addRate(newCurrency)
    }
    currencyRate.rates[newCurrency] = newRate
```

2. you would add it to the select list via `addRate`


```
function addRate(newCurrency) {
  const listofCurr = document.getElementById('curr_output')

  const listItem = document.createElement('option')
  listItem.value = newCurrency
  listItem.innerHTML = newCurrency
  listofCurr.appendChild(listItem)
}
```

As I mentioned, there are multiple ways how to achieve this :) So feel free to try another approach if you want :)